### PR TITLE
Add MacOS artefacts

### PR DESCRIPTION
**Pull request by fjtrujy**
### Description of Changes
This PR just basically generate artefacts for MacOS version and upload them to the action's artefacts.

### Rationale behind Changes
These changes were done to be able to download MacOS PCSX2 version from the GitHub Actions

Thanks

### CODE REVIEW

**Comment by tellowkrinkle** [comment]
I'd rather not have people download artifacts and then complain that they don't work because pcsx2 master has no GS backends for macOS

File: .github/workflows/macos-workflow.yml [hunk 1]
```diff
@@ -111,4 +111,20 @@ jobs:
 
       - name: Build PCSX2
         working-directory: build
-        run: make -j$(getconf _NPROCESSORS_ONLN) # macOS doesn't use make install
+        run: make -j$(getconf _NPROCESSORS_ONLN)
+
+      - name: Compress pcsx2 folder
+        working-directory: build/pcsx2
+        run: |
+          tar -zcvf pcsx2.tar.gz PCSX2.app
```
Comment (Margen67) [inline .github/workflows/macos-workflow.yml:119]:
```suggestion
        run: tar -C build/pcsx2 -zcvf pcsx2.tar.gz PCSX2.app
```

File: .github/workflows/macos-workflow.yml [hunk 2]
```diff
@@ -111,4 +111,20 @@ jobs:
...
+      - name: Get short SHA and ARCH
+        id: slug
+        run: |
+          echo "::set-output name=sha8::$(echo ${GITHUB_SHA} | cut -c1-8)"
+          echo "::set-output name=arch::$([ "${{ matrix.platform }}" == x86 ] && echo "32bit" || echo "64bit")"
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: PCSX2-${{ steps.slug.outputs.arch }}-[${{ steps.slug.outputs.sha8 }}]
+          path: build/pcsx2/pcsx2.tar.gz
```
Comment (Margen67) [inline .github/workflows/macos-workflow.yml:130]:
```suggestion
          path: build/pcsx2/pcsx2.tar.gz
          if-no-files-found: error
```

**Comment by fjtrujy** [comment]
@tellowkrinkle I understand your point if you want we can put something in the file name to make reference to that issue, as "incomplete"

**Comment by tellowkrinkle** [comment]
So what is the use of a macOS download that can't do anything?

**Comment by fjtrujy** [comment]
In my case it is useful because I'm downloading it and later on from your fork I' downloading the GS backend, and join both things

**Comment by tellowkrinkle** [comment]
That doesn't work anymore after GS merge

**Comment by lightningterror** [comment]
I will close this as tellow stated there is no GS backends for macOS, and GS has been merged in to core.

**Pull request closed without merging**